### PR TITLE
test: reproduce the issue with the cstr parsing inside a the proc macro

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
While I am working on trying to patch https://github.com/rust-lang/rust/issues/112820

I think it is useful to have a test for this case as known-bug

### CODE REVIEW

**Review by WaffleLapkin — APPROVED** [review]
sure

**Comment by compiler-errors** [comment]
Shouldn't this test be asserting that `c""` formats like `c""`?

Also, I have no idea why this needs to be committed as a separate test. The fix should be a minor fix to the proc macro library to handle cstrings correctly.

File: tests/ui/rfcs/rfc-3348-c-string-literals/auxiliary/wrong_parsing.rs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+extern crate proc_macro;
+
+use proc_macro::{TokenStream, TokenTree};
+
+#[proc_macro]
+pub fn repro(input: TokenStream) -> TokenStream {
+    for token in input {
+        if let TokenTree::Literal(literal) = token {
+            assert!(format!("{}", literal).contains(&"b\""), "panic on: `{}`", literal);
```
Comment (compiler-errors) [inline tests/ui/rfcs/rfc-3348-c-string-literals/auxiliary/wrong_parsing.rs:13]: This should be checking that a cstring starts with `c\"`?

**Comment by compiler-errors** [comment]
@bors r-

please fix the test so that it will eventually work when `c""` gets formatted correctly

File: tests/ui/rfcs/rfc-3348-c-string-literals/auxiliary/wrong_parsing.rs (hunk 1 above)
Comment (vincenzopalazzo) [reply tests/ui/rfcs/rfc-3348-c-string-literals/auxiliary/wrong_parsing.rs:13]: Good catch, thanks I commit the wrong version of the test

**Comment by vincenzopalazzo** [comment]
>please fix the test so that it will eventually work when c"" gets formatted correctly

Thanks @compiler-errors  for the catch! I push the correct version now.

>Also, I have no idea why this needs to be committed as a separate test. The fix should be a minor fix to the proc macro library to handle cstrings correctly.

Currently macros now have a working group and this issue should be discussed before inside the wg and then patched as you suggested. All the macros change will require at least one review from the wg due to that claim that we did inside the MCP.

Due to the original issue, there was some attempt (maybe) to solve the issue with cstr (See https://github.com/rust-lang/rust/issues/112820#issuecomment-1664774276) I wrote a test to confirm it. Now this may be not useful, but IMHO it will be for the next person who will patch this issue and will find the test.

**Comment by WaffleLapkin** [comment]
@vincenzopalazzo I went ahead and opened a fix for the issue: #116124. Sorry for taking an issue you've been working on, but it looked like you did not made progress since June, while the issue is easy to fix, so it seemed better to just fix it right away. In general, please remove your assignment or ask for help if you are struggling to fix an issue, so someone can help you, or work on the issue.

**Comment by vincenzopalazzo** [comment]
I need to apologize to @**Michael Goulet (compiler-errors)**  for saying the following statement and everyone that understand tha I wanted to impose my review of one review of the wg on each PR

>Currently macros have a working group and this issue should be discussed before inside the wg and then patched as you suggested. All the macros change will require at least one review from the wg due to that claim that we did inside the MCP.

In my message, I used the term required drawing a parallel to how the diagnostic working group is consulted for every change in their subsystem (though it might not be a perfect comparison, as I don't think there's an example quite like the macros working group). They can voice their opinions on changes. If they feel the need to reject a change, they can discuss it directly in the PR. That was my original intent behind using the word "required". It seems I might have chosen an imprecise term.

---

In addition due that I am mentioned the MCP, the MCP statement that I am referring to is:

>An important role for the WG is to bootstrap the experience necessary to make deeper changes to the macro system and to review changes to the PR system.

To me, this suggests that, at a minimum, we should be notified within the PR to review it. This offers us the opportunities to:

- Share the working group's perspective on the PR.
- If the PR is crafted by someone well-versed (not just in the Rust compiler but in the broader domain), we can learn from their expertise. Such interactions would be valuable in building our experience and also do not solve the current problem that the compiler has with the macros work.

---

thanks @WaffleLapkin for the patch, I had a similar version on https://git.hedwing.dev/vincenzopalazzo/rust Happy that we converge on it

**Pull request closed without merging**